File: 백준/Silver/1213. 팰린드롬 만들기/팰린드롬 만들기.cc
#include <bits/stdc++.h>
using namespace std;

string a, ret;
int n, mid;
int cnt[26];

int main() {
    cin >> a;
    
    n = (int)a.size();
    
    // cnt
    for(char c : a) cnt[c - 'A']++;
    
    
    // isPelindrome
    int odd = 0;
    for(int i = 0; i < 26; i++) {
        if(cnt[i] % 2 == 1) {
            odd++;
            mid = i + 'A';
        }
    }
    
    if(odd >= 2) {
        cout << "I'm Sorry Hansoo";
        return 0;
    }
    
    // 여기까지 오면 펠린드롬 가능
    for(int i = 0; i < 26; i++) {
        for(int j = 0; j < cnt[i] / 2; j++) {
            ret += (char)(i + 'A');
        }
    }
    
    if(odd == 1) ret += mid;
    
    // 나머지 짝수 cnt (사전역순 배치)
    for(int i = 25; i >= 0; i--) {
        for(int j = 0; j < cnt[i] / 2; j++) {
            ret += (char)(i + 'A');
        }
    }
    
    cout << ret;
    
    return 0;
}

File: 백준/Silver/1213. 팰린드롬 만들기/팰린드롬 만들기.swift
import Foundation

var s: [Character] = Array(readLine()!)

var cnt: [Character: Int] = [:]

for a in s {
    cnt[a, default: 0] += 1
}

var flag = 0
var flag2: Bool = false
var mid: Character = " "
var ret: [Character] = []
for i in (UnicodeScalar("A").value...UnicodeScalar("Z").value).reversed() {
    let c = Character(UnicodeScalar(i)!)
    if(cnt[c] != nil) {
        if(cnt[c]! % 2 == 1) {
            flag += 1
            mid = c
            flag2 = true
            cnt[c]! -= 1
        }
        if flag == 2 { break }
        for _ in stride(from: 0, to: cnt[c]!, by: 2) {
            ret.append(c)
            ret.insert(c, at: 0)
        }
    }
}
if(flag2) {
    ret.insert(mid, at: ret.count / 2)
}
if flag == 2 {
    print("I'm Sorry Hansoo")
} else {
    print(String(ret))
}
